tests/suites/test_suite_ssl.function

```diff
@@ -953,6 +966,11 @@ int mbedtls_endpoint_init( mbedtls_endpoint *ep, int endpoint_type, int pk_alg,
     ret = mbedtls_endpoint_certificate_init( ep, pk_alg );
     TEST_ASSERT( ret == 0 );
 
+    TEST_EQUAL( mbedtls_ssl_conf_get_user_data_n( &ep->conf ), user_data_n );
+    mbedtls_ssl_conf_set_user_data_p( &ep->conf, ep );
+    TEST_EQUAL( mbedtls_ssl_get_user_data_n( &ep->ssl ), user_data_n );
+    mbedtls_ssl_set_user_data_p( &ep->ssl, ep );
+
```

Comment:
You set the _p values above, so presumably you also want to test them:
```
TEST_EQUAL( mbedtls_ssl_conf_get_user_data_p( &ep->conf ), ep );
TEST_EQUAL( mbedtls_ssl_get_user_data_p( &ep->ssl ), ep );
```

Reply:
Right, I'm already checking them at the end of the handshake. It's better because this way we'll detect it if they somehow get overwritten. I didn't do it this way for the `_n` functions because in a given test run, for a given structure, I can either test either `_n` or `_p` but not both, and I didn't want to add the complexity of parametrizing runs to select one.

Having unit tests even for these trivial functions did turn up a bug: serialization doesn't save the user data. I ended up deciding it was a feature (because I think many applications that use both serialization and user data would want to serialize the data that's behind the pointer/handle and not the pointer/handle itself).